### ISSUE TITLE
Server lag with getMotherId and getFatherId

### ISSUE THREAD

**Issue by LemADEC** [WildBamaBoy/minecraft-comes-alive#594]
We've noted that MCA entities are fairly laggy compared to other entity/mobs. When checking with a profiler, we've found that calls to getMotherId and getFatherId are very slow due to calls to java.lang.Integer.parseInt().
I would suggest to replace the data structure to use integer instead of String, so save a lot of CPU power.

Here's an extract from our 21mn sampling:
![image](https://cloud.githubusercontent.com/assets/4776032/12061351/d497243c-af83-11e5-8651-319e71e1ad99.png)
mca.entity.EntityHuman.onUpdate used 8.49% of the server CPU
out of which 4.71% are due those String to Integer conversion.
in other words, fixing this issue would double the AI execution speed.

**Comment by WildBamaBoy** [comment]
Nice catch - can't change the format of mother and father ID data at this point, but there's a couple of things I can do about this.

**Comment by LemADEC** [comment]
In the readFromNBT() method, you can check if the old tag exist and convert it to the new format. This will progressively convert existing worlds to the new format.

**Comment by WildBamaBoy** [comment]
I'm not willing to do that as it goes quite a bit deeper than that.

**Comment by WildBamaBoy** [comment]
That will help, will make a couple more changes and probably publish this.

**Comment by WildBamaBoy** [comment]
Published. Should work a lot better for you now. :smile:

**Comment by LemADEC** [comment]
Thanks!
